pkg/health/health.go

```diff
@@ -0,0 +1,69 @@
+package health
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+)
+
+type Observable interface {
+	Ready(includes, excludes []string, verbose bool) error
+}
+
+type HealthzHandler interface {
+	Observe(...Observable)
+	HandleReadyzCheck(*http.Request) error
+}
+
+type HandlerOption func(HealthzHandler)
+
+func Observe(observables ...Observable) HandlerOption {
+	return func(h HealthzHandler) {
+		h.Observe(observables...)
+	}
+}
+
+func NewHandler(name string, options ...HandlerOption) HealthzHandler {
+	h := &handler{name: name}
+	for _, o := range options {
+		o(h)
+	}
+	return h
+}
+
+type handler struct {
+	name string
+
+	observables []Observable
+	mu          sync.RWMutex
```

Comment:
what shared (access) data is this MUTEX protecting?

Reply:
The `observables` slice.

I usually try to group mutex and corresponding data structures it's intended to protect in one block of declarations, separated from the other (non related) data structures.

Reply:
I think that would be the `observables`